### DISCUSSION TITLE
limiting access to management ports to specific IP

### DISCUSSION THREAD

**Discussion by dmille6** [telekom-security/tpotce#1473] [Q&A]
hi, i'd like to open up elasticsearch 64298:9200 to my internal network, so i can run queries. .. i run the honeypot outside of my normal network.. so i'd like to limit access to a specific IP address.. i know how to do that with UFW.. but this tpot honeypot system doesnt use UFW.. (maybe uses bind?)

what would be the easiest way to limit access to 64290-64300 to one specific IP address? So i can open up the elasticsearch IP address to that network? 

(i hope that makes sense)

**Comment by feloxxx** [comment]
Hello, I have a similar requirement to yours. Have you found a solution? Thank you!

**Comment by Ganesh-403** [comment]
@dmille6 
T-Pot doesn't use **UFW** to manage its firewall rules. Instead, it configures its own networking and firewall rules (via Netfilter/nftables depending on the host), so adding UFW rules is generally **not recommended**, as they may conflict with T-Pot's generated rules.

If your goal is simply to allow **one trusted internal IP** to access the management ports (e.g. `64290–64300`, including Elasticsearch on `64298`), the safest approach is to add a firewall rule **before** the generic drop/reject rule that:

- **accepts** traffic from your trusted IP to the management ports, and
- leaves all other traffic blocked.

For example, with nftables the rule would conceptually look like:

```nft
ip saddr 192.168.1.100 tcp dport {64290-64300} accept
```

or, if you only need Elasticsearch:

```nft
ip saddr 192.168.1.100 tcp dport 64298 accept
```

This preserves the default security posture while allowing only your workstation to connect. The nftables documentation covers matching on source addresses and destination ports if you need to adapt the rule to your ruleset. :contentReference[oaicite:0]{index=0}

Before making changes, I'd recommend checking how your T-Pot installation currently manages firewall rules:

```bash
sudo nft list ruleset
```

or, on older systems:

```bash
sudo iptables -L -n -v
sudo iptables -t nat -L -n -v
```

One thing to keep in mind is that **manual firewall changes may be overwritten** by T-Pot during upgrades or when `tpotinit` regenerates the configuration. From what I can tell, there isn't documented support for configuring per-IP allowlists for the management ports through a dedicated T-Pot setting, so if you make manual changes, you'll want to document them and verify they persist after updates.

A couple of questions that would help determine the best approach:

1. Which version of **T-Pot** are you running?
2. Is your host using **nftables** (`nft list ruleset`) or the legacy **iptables** backend?
3. Do you only need remote access to **Elasticsearch (64298)**, or do you also need Kibana/Web UI and the other management ports?

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
